src/App.jsx

```diff
@@ -14,45 +19,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+function getPreparedGoods(goods, { sortField, reverseStatus }) {
+  let preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods = preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        case SORT_FIELD_NAME:
+          return good1.localeCompare(good2);
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseStatus) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverseStatus, setReverseStatus] = useState(false);
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer,
+    { sortField, reverseStatus },
+  );
+  const isGoodsChanged = (sortField !== '' || reverseStatus);
```

Comment:
```suggestion
  const isGoodsChanged = (sortField || reverseStatus);
```
You don't need to compare with an empty string.
Article about [truthy](https://developer.mozilla.org/en-US/docs/Glossary/Truthy) values, and about [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy)